ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/how-to/custom-templates.md

```diff
@@ -0,0 +1,47 @@
+<!--
+  ~ Copyright (c) 2023 Arista Networks, Inc.
+  ~ Use of this source code is governed by the Apache License 2.0
+  ~ that can be found in the LICENSE file.
+  -->
+
+# Custom Templates
+
+With Custom Templates for the `eos_cli_config_gen` role, the user can create additional CLI configuration for features that are not covered by the built-in templates.
+
+## Create a Custom Template
+
+Custom templates should be written in Jinja2.
+
+For details on how to create Jinja2 templates, see [Templating Jinja2](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_templating.html#templating-jinja2)
+
+Custom templates can utilize any AVD variable for variable substitution including additional structured configuration variables created by a custom template in the `eos_designs` role. For details on how to create additional structured configuration variables see [Custom Templates for eos_designs](../../../eos_designs/docs/how-to/custom-templates.md) 
```

Comment:
The simpler way to provide variables is to use `custom_structured_configuration_` or the various `structured_config` keys in the data models. There is another howto on that, and the vars will show up in the output of `eos_designs` so they can indeed be picked up by the custom template in `eos_cli_config_gen`.

Reply:
Amended in the latest push.